dist/streams.js

```diff
@@ -1,36 +1,39 @@
 import EventEmitter from './events.js';
 
-const encoder = new TextEncoder();
-const decoder = new TextDecoder();
-const metadataPattern = /^mc:-?\d+;$/;
-const finisherByte = 59; // ;
+const numToBytesInt32 = (num) => {
+  const numBytes = new ArrayBuffer(4);
+  const numView = new DataView(numBytes);
+  numView.setInt32(0, num);
+  return numBytes;
+};
+
+const numFromBytesInt32 = (buffer) => {
+  const numView = new DataView(buffer);
+  return numView.getInt32(buffer.length);
+};
+
+const metadataCreator = (streamId) => numToBytesInt32(streamId);
```

Comment:
You can simply create 2 functions:
**getStreamId
createStreamIdBuffer**
without numToBytesInt32 & numFromBytesInt32, because it's not a generic Int32 conversion, but a special 4 bytes creation for streamId, which won't be used anywhere else:

```suggestion
const createStreamIdBuffer = (streamId) => {
  const buffer = new ArrayBuffer(STREAM_ID_LENGTH);
  const view = new DataView(buffer);
  view.setInt32(0, streamId);
  return buffer;
};

const getStreamId = (buffer) => {
  const view = new DataView(buffer);
  return view.getInt32(0);
};
```